src/mbgl/renderer/layers/render_background_layer.cpp

```diff
@@ -152,4 +141,23 @@ optional<Color> RenderBackgroundLayer::getSolidBackground() const {
     return { evaluated.get<BackgroundColor>() * evaluated.get<BackgroundOpacity>() };
 }
 
+namespace {
+void addPatternIfNeeded(const std::string& id, const LayerPrepareParameters& params) {
+    if (!params.patternAtlas.getPattern(id)) {
+        if (auto* image = params.imageManager.getImage(id)) {
+            params.patternAtlas.addPattern(*image);
```

Comment:
should return value be checked?

Reply:
actually, It's checked already in `RenderBackgroundLayer::render()`